server/datastore/mysql/software.go

```diff
@@ -314,12 +314,27 @@ func selectSoftwareSQL(hostID *uint, opts fleet.SoftwareListOptions) (string, []
 	}
 
 	if opts.WithHostCounts {
+		subSelectCounts := dialect.From(goqu.I("aggregated_stats").As("shc")).Select(
+			"shc.id", "shc.json_value", "shc.updated_at",
+		).Where(goqu.I("shc.type").Eq("software_hosts_count"), goqu.I("shc.json_value").Gt(0)).
+			SelectAppend(
+				goqu.I("shc.json_value").As("hosts_count"),
+				goqu.I("shc.updated_at").As("counts_updated_at"),
+			)
+		subSelectListOpts := opts.ListOptions
+		switch subSelectListOpts.OrderKey {
+		case "hosts_counts", "counts_updated_at":
+			// all good, known columns, so we sort
+			subSelectCounts = appendListOptionsToSelect(subSelectCounts, opts.ListOptions)
```

Comment:
The added sorting is the optimization?

Reply:
Right, and the filtering. So in the dashboard, we sort the aggregated stats table, limit it to 8 rows, and that's what we join against.

Reply:
Hmm ok so I _think_ I understand, the sub-select is not the optimization, it's out of necessity to apply the `ListOptions` to the `aggregated_stats` SELECT **in addtion** to the complete SELECT? So it will e.g. apply a LIMIT to the select on aggregated_stats, making only a few rows available to the join with software?

Reply:
Exactly.